tests/support/pytest/helpers.py

```diff
@@ -533,6 +533,212 @@ def shell_test_false():
     return "/bin/false"
 
 
+@attr.s(kw_only=True, frozen=True)
+class FakeSaltExtension:
+    tmp_path_factory = attr.ib(repr=False)
+    name = attr.ib()
+    pkgname = attr.ib(init=False)
+    srcdir = attr.ib(init=False)
+
+    @srcdir.default
+    def _srcdir(self):
+        return self.tmp_path_factory.mktemp("src", numbered=True)
+
+    @pkgname.default
+    def _pkgname(self):
+        replace_chars = ("-", " ")
+        name = self.name
+        for char in replace_chars:
+            name = name.replace(char, "_")
+        return name
+
+    def __attrs_post_init__(self):
+        self._laydown_files()
+
+    def _laydown_files(self):
+        if not self.srcdir.exists():
+            self.srcdir.mkdir()
+        setup_py = self.srcdir.joinpath("setup.py")
+        if not setup_py.exists():
+            setup_py.write_text(
+                textwrap.dedent(
+                    """\
+            import setuptools
+
+            if __name__ == "__main__":
+                setuptools.setup()
+            """
+                )
+            )
+        setup_cfg = self.srcdir.joinpath("setup.cfg")
+        if not setup_cfg.exists():
+            setup_cfg.write_text(
+                textwrap.dedent(
+                    """\
+            [metadata]
+            name = {0}
+            version = 1.0
+            description = Salt Extension Test
+            author = Pedro
+            author_email = pedro@algarvio.me
+            keywords = salt-extension
+            url = http://saltstack.com
+            license = Apache Software License 2.0
+            classifiers =
+                Programming Language :: Python
+                Programming Language :: Cython
+                Programming Language :: Python :: 3
+                Programming Language :: Python :: 3 :: Only
+                Development Status :: 4 - Beta
+                Intended Audience :: Developers
+                License :: OSI Approved :: Apache Software License
+            platforms = any
+
+            [options]
+            zip_safe = False
+            include_package_data = True
+            packages = find:
+            python_requires = >= 3.5
+            setup_requires =
+              wheel
+              setuptools>=50.3.2
+
+            [options.entry_points]
+            salt.loader=
+              module_dirs = {1}
+              runner_dirs = {1}.loader:get_runner_dirs
+              wheel_dirs = {1}.loader:get_new_style_entry_points
```

Comment:
**suggestion**: should also add the negative cases,  e.g. module_dirs` when there's nothing there, or points to the wrong directory, stuff like that.

Reply:
Will do on a separate PR